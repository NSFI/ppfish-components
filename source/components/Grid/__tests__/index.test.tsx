import { mount } from 'enzyme';
import React from 'react';
import { Row, Col } from '..';

describe('Grid', () => {
  it('Row should render', () => {
    const wrapper = mount(<Row />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Col should render', () => {
    const wrapper = mount(<Col span={2} />);
    expect(wrapper).toMatchSnapshot();
  });

  // it('when typeof gutter is object', () => {
  //   const wrapper = mount(<Row gutter={{ xs: 8, sm: 16, md: 24 }} />);
  //   expect(wrapper.find('div').first().props().style).toEqual(
  //     expect.objectContaining({
  //       marginLeft: -12,
  //       marginRight: -12,
  //     }),
  //   );
  // });

  it('Row wrapped Col should render', () => {
    const wrapper = mount(
      <Row>
        <Col span={12}></Col>
        <Col span={12}></Col>
      </Row>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('when type is flex should render', () => {
    const wrapper = mount(
      <Row type="flex">
        <Col span={12}>1</Col>
        <Col span={12}>2</Col>
      </Row>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('when order is set should render correct', () => {
    const wrapper = mount(
      <Row type="flex">
        <Col span={6} order={4}>
          1 col-order-4
        </Col>
        <Col span={6} order={3}>
          2 col-order-3
        </Col>
        <Col span={6} order={2}>
          3 col-order-2
        </Col>
        <Col span={6} order={1}>
          4 col-order-1
        </Col>
      </Row>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
