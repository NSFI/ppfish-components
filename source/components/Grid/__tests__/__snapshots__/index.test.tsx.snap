// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid Col should render 1`] = `
<Col
  span={2}
>
  <div
    className="fishd-col-2"
  />
</Col>
`;

exports[`Grid Row should render 1`] = `
<Row
  gutter={0}
>
  <div
    className="fishd-row"
  />
</Row>
`;

exports[`Grid Row wrapped Col should render 1`] = `
<Row
  gutter={0}
>
  <div
    className="fishd-row"
  >
    <Col
      key=".0"
      span={12}
    >
      <div
        className="fishd-col-12"
      />
    </Col>
    <Col
      key=".1"
      span={12}
    >
      <div
        className="fishd-col-12"
      />
    </Col>
  </div>
</Row>
`;

exports[`Grid when order is set should render correct 1`] = `
<Row
  gutter={0}
  type="flex"
>
  <div
    className="fishd-row-flex"
  >
    <Col
      key=".0"
      order={4}
      span={6}
    >
      <div
        className="fishd-col-6 fishd-col-order-4"
      >
        1 col-order-4
      </div>
    </Col>
    <Col
      key=".1"
      order={3}
      span={6}
    >
      <div
        className="fishd-col-6 fishd-col-order-3"
      >
        2 col-order-3
      </div>
    </Col>
    <Col
      key=".2"
      order={2}
      span={6}
    >
      <div
        className="fishd-col-6 fishd-col-order-2"
      >
        3 col-order-2
      </div>
    </Col>
    <Col
      key=".3"
      order={1}
      span={6}
    >
      <div
        className="fishd-col-6 fishd-col-order-1"
      >
        4 col-order-1
      </div>
    </Col>
  </div>
</Row>
`;

exports[`Grid when type is flex should render 1`] = `
<Row
  gutter={0}
  type="flex"
>
  <div
    className="fishd-row-flex"
  >
    <Col
      key=".0"
      span={12}
    >
      <div
        className="fishd-col-12"
      >
        1
      </div>
    </Col>
    <Col
      key=".1"
      span={12}
    >
      <div
        className="fishd-col-12"
      >
        2
      </div>
    </Col>
  </div>
</Row>
`;
